BlueprintUICommonControls/Sources/PointerInteraction.swift

```diff
@@ -0,0 +1,303 @@
+//
+//  PointerInteraction.swift
+//  BlueprintUICommonControls
+//
+//  Created by Kyle Van Essen on 6/26/20.
+//
+
+import UIKit
+import BlueprintUI
+
+
+
+/// An `Element` which lets you add `UIPointerInteraction` interactions
+/// to the contained `Element`.
+///
+/// A `UIPointerInteraction` allows you to respond to hover events for buttons,
+/// text, icons, etc, when a user is using an iPad that has support for a pointer, usually from a trackpad.
+///
+/// **Note** – On iOS versions before 13.4, this has no effect.
+///
+public struct PointerInteraction : Element
+{
+    /// The element to which the pointer action will be applied.
+    public var wrapping : Element
+    
+    /// How to provide the style based on the coordinate space and pressed keys.
+    public typealias StyleProvider = (UICoordinateSpace, UIKeyModifierFlags) -> Style
```

Comment:
Could we design this so you don't need a closure/builder unless you're actually building a style that needs those arguments? so you can do, e.g., 

```swift
element.pointerInteraction(style: .hidden)
element.pointerInteraction(style: .shape { coordinates, flags in
    // do the thing
})
```

it's small but nice IMO. If it seems weird/annoying to shape everything so this works, don't worry about it

Reply:
Ahh true, good call – I can add a helper method for that – should be just a couple lines